target/linux/samsung/patches-4.14/002-dm9000-clk.patch

```diff
@@ -0,0 +1,40 @@
+Index: linux-4.14.34/drivers/net/ethernet/davicom/dm9000.c
+===================================================================
+--- linux-4.14.34.orig/drivers/net/ethernet/davicom/dm9000.c
++++ linux-4.14.34/drivers/net/ethernet/davicom/dm9000.c
+@@ -39,6 +39,7 @@
+ #include <linux/regulator/consumer.h>
+ #include <linux/gpio.h>
+ #include <linux/of_gpio.h>
++#include <linux/clk.h>
+ 
+ #include <asm/delay.h>
+ #include <asm/irq.h>
+@@ -1436,6 +1437,7 @@ dm9000_probe(struct platform_device *pde
+ 	enum of_gpio_flags flags;
+ 	struct regulator *power;
+ 	bool inv_mac_addr = false;
++    const char *clk_name;
+ 
+ 	power = devm_regulator_get(dev, "vcc");
+ 	if (IS_ERR(power)) {
+@@ -1576,6 +1578,19 @@ dm9000_probe(struct platform_device *pde
+ 	/* fill in parameters for net-dev structure */
+ 	ndev->base_addr = (unsigned long)db->io_addr;
+ 
++    /* Enable clock if specified */
++    if (!of_property_read_string(dev->of_node, "clock-names", &clk_name)) {
++        struct clk *clk = devm_clk_get(dev, clk_name);
```

Comment:
Please just use a constant for the clock name like 
```
struct clk *clk = devm_clk_get(dev, "sromc");
```
and then handle the case that you get an -ENOENT back